src/test/java/io/github/w7mike/logic/ProjectsServiceTest.java

```diff
@@ -105,4 +130,59 @@ private JobConfigurationProperties configurationReturning(final boolean result)
         when(mockProperties.getTemplate()).thenReturn(mockTemplate);
         return mockProperties;
     }
+
+    private InMemoryGroupRepository inMemoryGroupRepository(){
+        return new InMemoryGroupRepository() {
+
+        };
+    }
+
+    private static class InMemoryGroupRepository implements JobGroupsRepository{
+
+        private Integer index = 0;
+        private Map<Integer, JobGroups> map = new HashMap<>();
+
+        public int count(){
+            return map.values().size();
+        }
+
+
+        @Override
+        public List<JobGroups> findAll() {
+            return new ArrayList<>(map.values());
+        }
+
+        @Override
+        public Optional<JobGroups> findById(final Integer id) {
+            return Optional.ofNullable(map.get(id));
+        }
+
+        @Override
+        public JobGroups save(final JobGroups entity) {
+            if (entity.getId() == 0){
+                try {
+                    JobGroups.class.getDeclaredField("id").set(entity, ++index);
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    throw new RuntimeException();
```

Comment:
it's a good practice to provide any information to the caller by filling `RuntimeException` with some useful info about the exception cause. E.g:
```
throw new RuntimeException(e.getMessage(), e);
```
This way, this exception will carry an info about its cause.